agent.go

```diff
@@ -392,7 +396,8 @@ func (a *Agent) connectivityChecks() {
 		updateInterval(a.disconnectedTimeout)
 		updateInterval(a.failedTimeout)
 
-		t := time.NewTimer(interval)
+		t.Reset(interval)
```

Comment:
There is a path where the previous timer has not been reset properly. If `a.forceCandidateContact` was set.

we'd need to do this here before resetting

```
if !t.Stop() {
	<-t.C
}
```

Reply:
There is a `Stop` call as soon as the timer is created ☝️ . So, a `Reset` can be applied any time after that. There are a couple of `Stop` calls 👇 , but I guess they become no-ops. Did I misread it?

Reply:
actually I've misread 🤦. the there is a Stop there as well. all good here :)